scheduler/src/cook/mesos/util.clj

```diff
@@ -552,3 +552,28 @@
   [ch ch-size]
   (->> (repeatedly ch-size #(async/poll! ch))
        (remove nil?)))
+
+(defn xform-pipe
```

Comment:
I like that we factored this out.  Can it have a better name?  Maybe `reduce-pipe`?  I want to capture that there is state inside this pipe.